src/aws/lambdaWithApiGateway/interface.ts

```diff
@@ -22,4 +22,9 @@ export interface ILambdaWithApiGateway extends ILambda {
    * Custom domain name for the API Gateway endpoint (optional)
    */
   customDomainName?: ICustomDomainName;
+
+  /**
+   * The protocol for the API Gateway endpoint (optional)
+   */
+  apiGatewayProtocolType?: string;
```

Comment:
[Valid values](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/apigatewayv2_api?lang=typescript#protocolType) can only be "HTTP" and "WEBSOCKET"
it would be better to use a type like type apiGWProtocolType = "HTTP" | "WEBSOCKET"
this would help LSP in autocomplete and will also warn the user, if user mis spells something.

Reply:
Added the type and updated the interface